app/src/main/kotlin/at/bitfire/davdroid/util/PermissionUtils.kt

```diff
@@ -73,6 +86,75 @@ object PermissionUtils {
                 locationAvailable
     }
 
+    /**
+     * Checks whether all conditions to access the current WiFi's SSID are met:
+     *
+     * 1. location permissions ([WIFI_SSID_PERMISSIONS]) granted (Android 8.1+)
+     * 2. location enabled (Android 9+)
+     *
+     * @return *true* if SSID can be obtained; *false* if the SSID will be <unknown> or something like that
+     */
+    @Composable
+    @ExperimentalPermissionsApi
+    fun canAccessWifiSsidLive(): State<Boolean> {
+        // If preview, cannot access WiFi SSID
+        if (LocalInspectionMode.current)
+            return remember { derivedStateOf { false } }
+
+        // before Android 8.1, SSIDs are always readable
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.O_MR1)
+            return remember { derivedStateOf { true } }
+
+        val context = LocalContext.current
+
+        val locationAvailable = MutableStateFlow(
+            // Android <9 doesn't require active location services, otherwise set initial state to false
+            Build.VERSION.SDK_INT < Build.VERSION_CODES.P
+        )
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
+            // Only if Android >= 9, we need to check for location services, so add listener
+            fun updateState(context: Context) {
+                val locationManager = context.getSystemService(Context.LOCATION_SERVICE) as LocationManager
```

Comment:
Location service status is not covered by `PROVIDERS_CHANGED_ACTION` (like gps/network provider), but by `MODE_CHANGED_ACTION`.